File: frontend/demo/component/combobox/combo-box-auto-open.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import '@vaadin/combo-box';
import { getCountries } from 'Frontend/demo/domain/DataService';
import type Country from 'Frontend/generated/com/vaadin/demo/domain/Country';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('combo-box-auto-open')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  @state()
  private items: Country[] = [];

  protected override async firstUpdated() {
    this.items = await getCountries();
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-combo-box
        auto-open-disabled
        label="Country"
        item-label-path="name"
        item-value-path="id"
        .items="${this.items}"
      ></vaadin-combo-box>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/combobox/ComboBoxAutoOpen.java
package com.vaadin.demo.component.combobox;

import com.vaadin.flow.component.combobox.ComboBox;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.demo.domain.Country;
import com.vaadin.demo.domain.DataService;

@Route("combo-box-auto-open")
public class ComboBoxAutoOpen extends Div {

  public ComboBoxAutoOpen() {
    // tag::snippet[]
    ComboBox<Country> comboBox = new ComboBox<>("Country");
    comboBox.setAutoOpen(false);
    add(comboBox);
    // end::snippet[]
    comboBox.setItems(DataService.getCountries());
    comboBox.setItemLabelGenerator(Country::getName);
  }

  public static class Exporter extends DemoExporter<ComboBoxAutoOpen> { // hidden-source-line
  } // hidden-source-line
}
